### PR TITLE
feat: acquisition royale subtabs categories

### PR DESCRIPTION
**Pull request by chrisling-dev**
Merge only after enterprises card are put into boxes, and Buy on Uniswap button is within a box

https://www.notion.so/FE-Project-Management-d3878c29e9534234a50f4f8838525bee?p=9b20ba4e1bd1486bb06ce14589629ba0

### CODE REVIEW

File: apps/frontend/acquisition-royale-home/src/components/MainTab.tsx [hunk 1]
```diff
@@ -1,42 +1,100 @@
...
+  content: '';
+  display: inline-block;
+  height: 0;
+  left: 50%;
+  position: absolute;
+  transform: translateX(-50%);
+  width: 0;
+`
+
+const Wrapper = styled.div<{ showSubtabArrow: boolean }>`
```
Comment (jonezque) [inline apps/frontend/acquisition-royale-home/src/components/MainTab.tsx:24]: ensure there's no warning,
`$showSubtabArrow`

File: apps/frontend/acquisition-royale-home/src/components/Subtabs.tsx [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+    <SubTabPaneWrapper>{children}</SubTabPaneWrapper>
+  </ATabPane>
+)
+
+const Subtabs: React.FC<TabsProps & { subtabs: SubtabProps[] }> = ({ subtabs, ...props }) => (
+  <Wrapper>
+    {/* eslint-disable-next-line react/jsx-props-no-spreading */}
+    <Tabs centered animated={{ inkBar: false }} {...props}>
+      {subtabs.map(({ content, tab }) => (
+        <SubTabPane tab={tab} key={tab.toLowerCase()}>
```
Comment (jonezque) [inline apps/frontend/acquisition-royale-home/src/components/Subtabs.tsx:72]: Do we really need `.toLowerCase() `here?

File: apps/frontend/acquisition-royale-home/src/features/mvp/GamePage.tsx [hunk 3]
```diff
@@ -54,44 +36,13 @@ const GamePage: React.FC = () => {
...
-          </CenterWrapper>
-        </TabPane>
-        <TabPane tab="CUSTOMIZE" key="customize">
-          <CenterWrapper>
-            <Rename />
-          </CenterWrapper>
-        </TabPane>
+      <MainTab subtabs={['rp', 'play']} defaultActiveKey="rp" centered>
+        {categories.map(({ tab, content, subtabs }) => (
+          <TabPane tab={tab} key={tab.toLowerCase()}>
```
Comment (jonezque) [inline apps/frontend/acquisition-royale-home/src/features/mvp/GamePage.tsx:41]: ditto

File: apps/frontend/acquisition-royale-home/src/components/MainTab.tsx [hunk 4]
```diff
@@ -1,42 +1,100 @@
...
-    {/* eslint-disable-next-line react/jsx-props-no-spreading */}
-    <Tabs animated={{ inkBar: false }} {...props} />
-  </Wrapper>
-)
+const MainTab: React.FC<TabsProps & { subtabs?: string[] }> = ({
+  defaultActiveKey,
+  subtabs = [],
+  ...props
+}) => {
+  const [activeKey, setActiveKey] = useState<string>(defaultActiveKey)
```
Comment (jonezque) [inline apps/frontend/acquisition-royale-home/src/components/MainTab.tsx:86]: I think `<string>` can be inferred

**Review by jonezque — APPROVED** [review]
Great Job!
left minor comments, not blocking  approval

File: apps/frontend/acquisition-royale-home/src/components/MainTab.tsx (hunk 1 above)
Comment (chrisling-dev) [reply apps/frontend/acquisition-royale-home/src/components/MainTab.tsx:24]: nice catch!

File: apps/frontend/acquisition-royale-home/src/components/Subtabs.tsx (hunk 2 above)
Comment (chrisling-dev) [reply apps/frontend/acquisition-royale-home/src/components/Subtabs.tsx:72]: cause MainTab is taking `subtabs` to check whether the ui need to show the subtab arrow. I thought passing all small cap looks cleaner, hence keeping it consistent even for lowerCase, but I'm going to remove this cause I recall an improvement I have in mind hehe

**Pull request merged**